db/migrate/20161006070258_migrate_common_parameter_permissions.rb

```diff
@@ -0,0 +1,29 @@
+class MigrateCommonParameterPermissions < ActiveRecord::Migration
+  class FakeFilter < ActiveRecord::Base
+    self.table_name = 'filters'
+  end
+
+  def up
+    all_new_permissions = Permission.where(:name => ['view_params', 'edit_params', 'create_params', 'destroy_params'])
+
+    affected = Filter.includes(:permissions).where(:permissions => {:name => ['view_globals', 'edit_globals', 'create_globals', 'destroy_globals']})
+    affected.each do |filter|
+      new_names = filter.permissions.map { |old| old.name.sub(/globals/, 'params') }
+      new_permissions = all_new_permissions.select do |new|
+        new_names.include?(new.name)
+      end
+
+      fake_filter = FakeFilter.find(filter.id)
+      new_search = "type = CommonParameter"
+      new_search = filter.search + ' and ' + new_search if filter.search.present?
```

Comment:
`filter.search` should probably be wrapped in parentheses (inside the search string) in case it's an `or` expression or something, to keep precedence clear + correct